docs/reference/modules/module.md

```diff
@@ -0,0 +1,103 @@
+
+Templates rendered and sent with this method have full access to all template [data properties](/guide/template-data.md) appropriate for the related module. The template is also wrapped with the proper layout file (`'@apostrophecms/template:outerLayout.html'` by default), including the full `head` tag.
+
+This method triggers the `@apostrophecms/page` module to emit a [`beforeSend` event](/reference/server-events.md#beforesend).
+
+### `enableBrowserData(scene)`
+
+Call this method from a module's [initialization function](/reference/module-api/module-overview.md##initialization-function) if the module implements the [`getBrowserData()` method](#getbrowserdata-req). **This is only necessary for modules that do not already do this** (all doc type and widget type modules already do this). The data returned by `getBrowserData(req)` will then be available on `apos.modules['the-module-name']` in the browser.
+
+The `scene` argument is only needed if making the data available while logged-out. By default browser data is pushed only for the `apos` scene (the logged-in context), so the anonymous site visitor experience will include the extra data (except on the /login page and other pages that opt into the `apos` scene). If `scene` is set to `public` then the data is available to all visitors.
+
+### `getBrowserData(req)`
+
+The `getBrowserData()` method can be extended to adjust or add data that the module makes available in the browser. It should return an object to make the data available in browsers (as controlled by [the `enableBrowserData()` method](#enablebrowserdata-scene)).
+
+The data will be available on a property of the `apos.modules` object matching the module name. For example, `@apostrophecms/i18n` module data is available on `apos.modules['@apostrophecms/i18n']`. If the module has an alias the data will also be accessible via `apos.yourAlias`.
+
+Many Apostrophe core modules already populate browser data, including piece type, page type, and widget type modules. For this reason, **it is ususally correct to *extend* this method to add additional browser data** using the module's [`extendMethods` customization function](/reference/module-api/module-overview.md#extendmethods-self).
+
+Avoid returning large data structures, as this will impact page load time and performance.
+
+### `getOption(req, dotPathOrArray, def)`
+
+This is a convenience method to fetch properties of `self.options` in a module. It will look for, then return the option identified by the second argument.
+
+`req` is required to provide extensibility. Modules can use it to change the response based on the current page and other factors tied to the request. The second argument, `dotPathOrArray`, may be a dot path (`'flavors.grape.sweetness'`) or an array `[ 'flavors', 'grape', 'sweetness' ]`.
```

Comment:
Modules that override this method can use it...

Reply:
Actually I think we're going to deprecate use of req in `getOption`. It underpins apostrophe-override-options in A2 but we've successfully advocated so far for replacing that with just a way for the global doc to inject options into modules, which would happen by recreating the apos object like in multisite, so no need for this messiness.

Reply:
Since the key is always a string we should be able to introduce a "just pass the key and don't pass req at all" variation too once we commit on that.

Reply:
> Modules that override this method can use it...

I'm not sure what that means.

Reply:
> Actually I think we're going to deprecate use of req in getOption.

We can remove this for now, comment it out, or leave it for and update when that's deprecated. What do you prefer?